lib/workers.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const P = require('./promise');
+
+const AppError = require('./error');
+const config = require('./config');
+const logger = require('./logging')('workers');
```

Comment:
Naming nit: I feel like "workers" is too generic a name here.  Unless we expect this to grow responsibilities beyond handling avatars, perhaps we should name it something more specific, so that calling code reads like `avatarHandler.upload() or maybe `workers.uploadAvatar()` rather than just `workers.upload()` which doesn't provide much information.